packages/plugins/other/visitor-plugin-common/src/selection-set-to-object.ts

```diff
@@ -112,8 +121,18 @@ export class SelectionSetToObject<Config extends ParsedDocumentsConfig = ParsedD
         const spreadsUsage = this.buildFragmentSpreadsUsage(spreads);
         const directives = (node.directives as DirectiveNode[]) || undefined;
 
+        // When we collect the selection sets of inline fragments we need to
+        // make sure directives on the inline fragments are stored in a way
+        // that can be associated back to the fields in the fragment, to
+        // support things like making those fields optional when deferring a
+        // fragment (using @defer).
```

Comment:
This alludes to the biggest change in this PR. When the `buildSelectionSet` function is trigged in this file, we need a way to associate fields in the selection set back to directives that might have been set on the fragment containing those fields. I'm doing this by adding a new `fragmentDirectives` property to the collected fields that holds any fragment directives that were specified, so when we later need to decide if a field should be marked as optional, we can look into its `fragmentDirectives` property to see if something like `@defer` was used. This approach also makes sure only the directives defined on the fragment related to a field are used with that field specifically, which means multiple inline fragments with different/no directives can now be properly handled.